superset/hive_query.py

```diff
@@ -17,25 +35,46 @@
 }
 
 # return seconds from gran valaue map default is 86400
-def get_gran_value_in_seconds(value):
-    if  value in GRAN_VALUE_MAP:
+def get_gran_value_in_seconds(value,time_partitions):
+    partition_min_grain = get_partitions_min_grain(time_partitions)
+    if value in GRAN_VALUE_MAP and  GRAN_VALUE_MAP[value] >= partition_min_grain:
         return GRAN_VALUE_MAP[value]
-    return 86400
+    return partition_min_grain
```

Comment:
I think we should always use partition-min-gran and ignore the provided value as data will come on min grain only.

Reply:
user can choose higher grain and we should honor that creating date points

Reply:
Lets document the current behaviour with an example

Reply:
If some choose higher grain  Query will be formed at  higher grain interval but up to minimum  partition supported by table as per selected time  and  data will fetch for minimum partitioned bin-interval  on every selected grain interval and it will create data mismatch race condition. and this condition is open only for to serve only this type of use case like see data at 10-11 hr bin interval for 7 days